test/api/test_tools.py

```diff
@@ -802,6 +802,21 @@ def test_run_cat1_with_two_inputs(self, history_id):
         output1_content = self.dataset_populator.get_history_dataset_content(history_id, dataset=output1)
         self.assertEqual(output1_content.strip(), "Cat1Test\nCat2Test")
 
+    @skip_without_tool("mapper_two")
+    @uses_test_history(require_new=False)
+    def test_bam_state_regression(self, history_id):
+        new_dataset1 = self.dataset_populator.new_dataset(history_id, content='123\n456\n789')
+        inputs = {
+            'input1': dataset_to_param(new_dataset1),
+            'reference': dataset_to_param(new_dataset1),
+        }
+        outputs = self._run_and_get_outputs('mapper_two', history_id, inputs)
+        # Assert we have three outputs with 1, 2, and 3 lines respectively.
```

Comment:
There are 4 outputs, and also the rest of the comment seems wrong.

Reply:
Yes - must have been copy-paste problems. Thanks for the catch.